src/main/java/Parser.java

```diff
@@ -0,0 +1,124 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+
+    /**
+     * Parse a string to Todo.
+     * @param string the string.
+     * @return the todo needed.
+     *
+     */
+
+    public static Todo stringToTodo(String string) {
+        String desc = "";
+        String[] texts = string.split("\\s+");
+        boolean status = (texts[0].charAt(4) == 'X');
+        for (int j = 1; j < texts.length ; j++) {
+            desc = desc + texts[j] + " ";
+        }
+        return new Todo(desc,status);
+    }
+
+
+    /**
+     * Parse a string to Deadline.
+     * @param string the string.
+     * @return the deadline needed.
+     *
+     */
+
+
+    public static Deadline stringToDeadline(String string){
+        String missions = "";
+        String[] time = string.split("[|]");
+        String[] texts = string.split("\\s+");
+        String content = time[0];
+        String[] text = content.split("\\s+");
+        boolean status = (texts[0].charAt(4) == 'X');
+        for (int j = 1; j < text.length ; j++) {
+            missions = missions + texts[j] + " ";
+        }
+        String[] times = time[1].split("\\s+");
+        return new Deadline(missions,status
+                , LocalDate.parse(times[1], DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+                , LocalTime.parse(times[2],DateTimeFormatter.ofPattern("HH:mm")));
+    }
+
+    /**
+     * Parse a string to Event.
+     * @param string the string.
+     * @return the event needed.
+     *
+     */
+
+    public static Event stringToEvent(String string){
+        String missions = "";
+
+        String[] time = string.split("[|]");
+        String[] texts = string.split("\\s+");
+        String content = time[0];
+        String[] text = content.split("\\s+");
+        for (int j = 1; j < text.length ; j++) {
+            missions = missions + texts[j] + " ";
+        }
+        String[] startEnd = time[1].split("[~]");
+        String[] start = startEnd[0].split("\\s+");
+        String[] end = startEnd[1].split("\\s+");
+        boolean status = (texts[0].charAt(4) == 'X');
+        return new Event(missions,status,
+                LocalDate.parse(start[1],DateTimeFormatter.ofPattern("yyyy-MM-dd")),
+                LocalTime.parse(start[2],DateTimeFormatter.ofPattern("HH:mm")),
+                LocalDate.parse(end[1],DateTimeFormatter.ofPattern("yyyy-MM-dd")),
+                LocalTime.parse(end[2],DateTimeFormatter.ofPattern("HH:mm")));
+    }
```

Comment:
As a suggestion, perhaps define these `DataTimeFormatter` patterns as `static` and `final` constants as they seem to be used quite a number of time throughout your code.